pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -265,15 +266,44 @@ func validatePipelineResults(results []PipelineResult) (errs *apis.FieldError) {
 		expressions = filter(expressions, looksLikeResultRef)
 		resultRefs := NewResultRefs(expressions)
 		if len(expressions) != len(resultRefs) {
-			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
+			return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("expected all of the expressions %v to be result expressions but only %v were", expressions, resultRefs),
 				"value").ViaFieldIndex("results", idx))
 		}
 
+		if !TaskContainsResult(result.Value, pipelineTaskNames) {
+			return errs.Also(apis.ErrInvalidValue("referencing a nonexistent task",
+				"value").ViaFieldIndex("results", idx))
+		}
 	}
 
 	return errs
 }
 
+// put task names in a set
+func getPipelineTasksNames(pipelineTasks []PipelineTask) sets.String {
+	pipelineTaskNames := make(sets.String)
+	for _, pipelineTask := range pipelineTasks {
+		pipelineTaskNames.Insert(pipelineTask.Name)
+	}
+
+	return pipelineTaskNames
+}
+
+// TaskContainsResult ensures the result value is referenced within the
+// task names
+func TaskContainsResult(value string, pipelineTaskNames sets.String) bool {
```

Comment:
this function is used within this package only so it should be unexported

```suggestion
func taskContainsResult(value string, pipelineTaskNames sets.String) bool {
```

Reply:
thank you for the suggestion, done.